repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java

```diff
@@ -124,6 +133,197 @@ public void retryTask(String taskGuid) throws AtlasBaseException {
+        return deletedTasks;
+    }
+
+    private void deleteTask(String taskGuid) throws AtlasBaseException {
+        DeleteType deleteType = RequestContext.get().getDeleteType();
+        if (deleteType == DeleteType.SOFT || deleteType == DeleteType.DEFAULT) {
+            softDelete(taskGuid);
+        } else if (deleteType == DeleteType.HARD) {
+            hardDelete(taskGuid);
+        }
+    }
+
+    @GraphTransaction
+    public AtlasVertex createTaskVertex(AtlasTask task) {
+        AtlasVertex ret = graph.addVertex();
+
+        setEncodedProperty(ret, Constants.TASK_GUID, task.getGuid());
+        setEncodedProperty(ret, Constants.TASK_TYPE_PROPERTY_KEY, Constants.TASK_TYPE_NAME);
+        setEncodedProperty(ret, Constants.TASK_STATUS, task.getStatus().toString());
+        setEncodedProperty(ret, Constants.TASK_TYPE, task.getType());
+        setEncodedProperty(ret, Constants.TASK_CREATED_BY, task.getCreatedBy());
+        setEncodedProperty(ret, Constants.TASK_CREATED_TIME, task.getCreatedTime());
+        setEncodedProperty(ret, Constants.TASK_UPDATED_TIME, task.getUpdatedTime());
+        if (task.getClassificationId() != null) {
+            setEncodedProperty(ret, Constants.TASK_CLASSIFICATION_ID, task.getClassificationId());
+        }
+
+        if(task.getEntityGuid() != null) {
+            setEncodedProperty(ret, Constants.TASK_ENTITY_GUID, task.getEntityGuid());
+        }
+
+        if (task.getStartTime() != null) {
+            setEncodedProperty(ret, Constants.TASK_START_TIME, task.getStartTime().getTime());
+        }
+
+        if (task.getEndTime() != null) {
+            setEncodedProperty(ret, Constants.TASK_END_TIME, task.getEndTime().getTime());
+        }
+
+        setEncodedProperty(ret, Constants.TASK_PARAMETERS, AtlasJson.toJson(task.getParameters()));
+        setEncodedProperty(ret, Constants.TASK_ATTEMPT_COUNT, task.getAttemptCount());
+        setEncodedProperty(ret, Constants.TASK_ERROR_MESSAGE, task.getErrorMessage());
+
+        LOG.info("Creating task vertex: {}: {}, {}: {}, {}: {} ",
+                Constants.TASK_TYPE, task.getType(),
+                Constants.TASK_PARAMETERS, AtlasJson.toJson(task.getParameters()),
+                TASK_GUID, task.getGuid());
+
+        return ret;
+    }
+
+    @GraphTransaction
+    @Override
+    public void hardDelete(String guid) throws AtlasBaseException {
+        try {
+            AtlasGraphQuery query = graph.query()
+                    .has(Constants.TASK_TYPE_PROPERTY_KEY, Constants.TASK_TYPE_NAME)
+                    .has(TASK_GUID, guid);
+
+            Iterator<AtlasVertex> results = query.vertices().iterator();
+
+            if (results.hasNext()) {
+                graph.removeVertex(results.next());
+            }
+        } catch (Exception exception) {
+            LOG.error("Error: deletingByGuid: {}", guid);
+
+            throw new AtlasBaseException(exception);
+        } finally {
+            graph.commit();
```

Comment:
Do we need to do graph.commit() manually?

Reply:
Added @Graphtransaction annotation